infoschema/perfschema/tables.go

```diff
@@ -237,11 +237,8 @@ func (vt *perfSchemaTable) getRows(ctx sessionctx.Context, cols []*table.Column)
 }
 
 // IterRecords implements table.Table IterRecords interface.
-func (vt *perfSchemaTable) IterRecords(ctx sessionctx.Context, startKey kv.Key, cols []*table.Column,
+func (vt *perfSchemaTable) IterRecords(ctx sessionctx.Context, cols []*table.Column,
 	fn table.RecordIterFunc) error {
-	if len(startKey) != 0 {
```

Comment:
We removed this check, but did not add it back. Does `perfSchemaTable` rely on this behavior?

Reply:
1. `perfSchemaTable`  does not rely on this behavior, as you can see, the following line just returns  `table.ErrUnsupportedOp`.
2. No code actually uses this parameter, the caller always passes a `[-inf,...]` range which means the whole table.
3. It's a poor abstract for `table.Table` interface, because not all tables have the concept of iterate by key-range(perf schema for example)

Reply:
Got